packages/core/strapi/lib/services/entity-validator/index.js

```diff
@@ -222,10 +223,134 @@ const createValidateEntity =
         entity,
       },
       { isDraft }
-    ).required();
+    )
+      .test('relations-test', 'check that all relations exist', async function (data) {
+        try {
+          await checkRelationsExist(buildRelationsStore({ uid: model.uid, data }));
+        } catch (e) {
+          return this.createError({
+            path: this.path,
+            message: e.message,
+          });
+        }
+        return true;
+      })
+      .required();
+
     return validateYupSchema(validator, { strict: false, abortEarly: false })(data);
   };
 
+/**
+ * Assign values in the relationsStore
+ * @param {Object} relationsValue containing the relation IDs
+ * @param {String} target the UID of these relations
+ * @param {Object} relationsStore to be updated
+ */
+const updateRelationsStore = ({ relationsValue = null, target = '', relationsStore = {} }) => {
+  if (isEmpty(target) || isNil(relationsValue)) {
+    return;
+  }
+  if (
+    typeof relationsValue === 'object' &&
+    (`count` in relationsValue || `disconnect` in relationsValue)
+  ) {
+    // this logic should not apply to relations being removed or counted
```

Comment:
I don't understand the `count` usecase. When creating or updating an entry there is no way to count no?

Reply:
Also it's possible to `connect` and `disconnect` at the same time. So we should not return earlier

Reply:
sorry yes I've just realised my mistake here 🤦‍♂️

Reply:
No problem! It's nice you had that in mind